pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java

```diff
@@ -531,12 +549,14 @@ private void initializeProfile(Handle handle, User user, UserRegistrationPayload
         String lastName = payload.getLastName();
         lastName = StringUtils.isNotBlank(lastName) ? lastName.trim() : null;
         long languageId = determineUserLanguage(handle, payload).getId();
+        ZoneId timeZone = parseUserTimeZone(payload.getTimeZone());
```

Comment:
Let's fail the request if the user input an invalid IANA code- I'm thinking a `400 BAD REQUEST` code. My justification for this is we are expecting a specific type of data- an IANA identifier, not a generic string. If it's optional, `null`ing it out, or excluding the value should be fine, but if an invalid identifier is sent we need to let the client know, especially since we aren't exposing the TZ anywhere. Having a silent failure that subtly modifies server behavior, _even deterministically_, creates a very unfriendly API.

Reply:
Added a check for it much earlier in the request.